src/thorlabs_cube/driver/kcube/kpa.py

```diff
@@ -0,0 +1,181 @@
+                self.status_report_counter = 0
+                await self.send(Message(MGMSG.QUAD_ACK_STATUSUPDATE))
+            else:
+                self.status_report_counter += 1
+
+        else:
+            raise MsgError(f"Unhandled message ID: {msg_id}")
+
+    async def set_trigger_config(
+        self,
+        trig1_mode: int,
+        trig1_polarity: int,
+        trig1_sum_min: int,
+        trig1_sum_max: int,
+        trig1_diff_threshold: int,
+        trig2_mode: int,
+        trig2_polarity: int,
+        trig2_sum_min: int,
+        trig2_sum_max: int,
+        trig2_diff_threshold: int,
+    ) -> None:
+        """Set trigger configuration for both TRIG1 and TRIG2.
+
+        :param trig1_mode: TRIG1 operating mode.
+        :param trig1_polarity: TRIG1 polarity.
+        :param trig1_sum_min: TRIG1 sum minimum.
+        :param trig1_sum_max: TRIG1 sum maximum.
+        :param trig1_diff_threshold: TRIG1 differential threshold.
+        :param trig2_mode: TRIG2 operating mode.
+        :param trig2_polarity: TRIG2 polarity.
+        :param trig2_sum_min: TRIG2 sum minimum.
+        :param trig2_sum_max: TRIG2 sum maximum.
+        :param trig2_diff_threshold: TRIG2 differential threshold.
+        """
+        payload = st.pack(
+            "<HHHHHHHHHHHHH",
+            trig1_mode,
+            trig1_polarity,
+            trig1_sum_min,
+            trig1_sum_max,
+            trig1_diff_threshold,
+            trig2_mode,
+            trig2_polarity,
+            trig2_sum_min,
+            trig2_sum_max,
+            trig2_diff_threshold,
+            _RESERVED,
+            _RESERVED,
+            _RESERVED,
+        )
+        await self.send(Message(MGMSG.QUAD_SET_PARAMS, data=payload))
+
+    async def get_trigger_config(
+        self,
+    ) -> tuple[int, int, int, int, int, int, int, int, int, int]:
+        """Get trigger configuration for both TRIG1 and TRIG2.
+
+        :return: A tuple containing trigger configuration parameters for TRIG1 and TRIG2.
+        """
+        get_msg = await self.send_request(
+            MGMSG.QUAD_REQ_PARAMS, [MGMSG.QUAD_GET_PARAMS], _CHANNEL
+        )
+        return st.unpack("<HHHHHHHHHHHHH", get_msg.data[2:27])
+
+    async def set_digital_outputs(self, digital_outputs: int) -> None:
+        """Set digital outputs for TRIG1 and TRIG2.
+
+        :param digital_outputs: Status of TRIG1 and TRIG2 outputs.
+        """
+        payload = st.pack(
+            "<HHH", SUBMSG.QUAD_KPA_DIGOPS_SUB_ID, digital_outputs, _RESERVED
+        )
+        await self.send(Message(MGMSG.QUAD_SET_PARAMS, data=payload))
+
+    async def get_digital_outputs(self) -> int:
+        """Get digital outputs for TRIG1 and TRIG2.
+
+        :return: Status of TRIG1 and TRIG2 outputs.
+        """
+        get_msg = await self.send_request(
+            MGMSG.QUAD_REQ_PARAMS, [MGMSG.QUAD_GET_PARAMS], _CHANNEL
+        )
+        return st.unpack("<HHH", get_msg.data[2:5])[0]
+
+
+class KpaSim:
```

Comment:
```suggestion
class KpaSim(TpaSim):
```
Is this inheritance needed too?